experimental/sql/src/ibis_to_alg.py

```diff
@@ -0,0 +1,101 @@
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+from dataclasses import dataclass
+from xdsl.ir import Operation, MLContext, Region, Block
+from typing import List, Type, Optional
+from xdsl.dialects.builtin import ArrayAttr, StringAttr, ModuleOp, IntegerAttr
+
+from xdsl.pattern_rewriter import RewritePattern, GreedyRewritePatternApplier, PatternRewriteWalker, PatternRewriter, op_type_rewrite_pattern
+
+import dialects.ibis_dialect as ibis
+import dialects.rel_alg as RelAlg
+
+
+@dataclass
+class IbisRewriter(RewritePattern):
+
+  def convert_datatype(self, type_: ibis.DataType) -> RelAlg.DataType:
+    if isinstance(type_, ibis.String):
```

Comment:
This could be done with a `multipledispatch` as well, no?

Reply:
I think this kind of code should ideally be implemented using the new python [switch statement](https://towardsdatascience.com/switch-case-statements-are-coming-to-python-d0caf7b2bfd3) ,but since xDSL and our entire repo does not yet run on python 3.10, I left this for the moment.

Reply:
Language support for pattern matching will be awesome, I am really looking forward to that feature!